catalog/ui/search-ui/standard/src/main/webapp/js/view/preferences/PreferencesModal.view.js

```diff
@@ -14,37 +14,227 @@
+            tagName: 'tr',
+            className: 'layerPicker-row',
             events: {
-                'hidden.bs.modal': 'saveChanges'
+                'click input[type="range"]': 'changeAlpha',
+                'click input[type="checkbox"]': 'changeShow',
+                'click button[data-value="up"]': 'changeUp',
+                'click button[data-value="down"]': 'changeDown'
+            },
+            initialize: function (options) {
+                this.viewOptions = options;
+                Handlebars.registerHelper('canRaise', function () {
+                    var index = options.collection.indexOf(options.model);
+                    return index < options.collection.length - 1 ? '' : 'pickerNoRaise';
+                });
+                Handlebars.registerHelper('canLower', function () {
+                    var index = options.collection.indexOf(options.model);
+                    return index > 0 ? '' : 'pickerNoLower';
+                });
+            },
+            changeAlpha: function () {
+                var newAlphaValue = this.$el.find('input[type="range"]').val();
+                this.model.set('alpha', newAlphaValue);
+
+                var layer = this.viewOptions.layerForCid[this.model.cid];
+                layer.alpha = newAlphaValue;
+            },
+            changeShow: function () {
+                var newShowValue = this.$el.find('input[type="checkbox"]').is(':checked');
+                this.model.set('show', newShowValue);
+
+                var layer = this.viewOptions.layerForCid[this.model.cid];
+                layer.show = newShowValue;
+            },
+            changeUp: function () {
+                var layer = this.viewOptions.layerForCid[this.model.cid];
+                this.viewOptions.getCesiumViewer().imageryLayers.raise(layer);
+                this.model.collection.moveUp(this.model);
+            },
+            changeDown: function () {
+                var layer = this.viewOptions.layerForCid[this.model.cid];
+                this.viewOptions.getCesiumViewer().imageryLayers.lower(layer);
+                this.model.collection.moveDown(this.model);
+            }
+        });
+
+        PrefsModalView.LayerPickers = Marionette.CollectionView.extend({
+            template: 'layerPickerViewTemplate',
+            childView: PrefsModalView.LayerPicker,
+            tagName: 'table',
+            className: 'table layerPickerTable',
+            childViewOptions: {},
+            initialize: function (options) {
+                // Marionette.CollectionView implicitly handles options.collection
+                this.childViewOptions = options; // pass options to child views.
+            },
+            /*
+             * following code is copy/paste of
+             * https://github.com/marionettejs/backbone.marionette/blob/master/src/collection-view.js#L579-L618
+             * but changed to "reverse" order of rendered list.
+             */
```

Comment:
I chose this implementation in the short term to get the prototype working.  The Backbone.Collection API suggested overriding the attachHtml() method in order to achieve a "prepend" rendering of the collection. Only later did I notice that the Cesium ImageryLayerCollection API allowed for specifying the index when adding the provider.  Overriding attachHtml() is a bit complex and this implementation should be replaced/simplified.

This problem presumes that the model collection has an implicit ordering derived from the order the imagery providers are added in the admin UI.  It might be better to require the admin to specify an explicit ordering in the config data.